packages/eslint-plugin/src/rules/no-redundant-type-constituents.ts

```diff
@@ -528,6 +943,33 @@ export default createRule({
             });
           }
         }
+
+        for (const [typeNode, typeRelations] of overriddenObjectTypes) {
+          const groupTypeRelationsBySuperTypeName =
+            getGroupTypeRelationsBySuperTypeName(typeRelations);
+
+          for (const [
+            superTypeName,
+            typeRelationAndNames,
+          ] of groupTypeRelationsBySuperTypeName) {
+            const mergedSubTypeName = mergeTypeNames(
+              typeRelationAndNames.map(({ subTypeName, subTypeNode }) => ({
+                typeName: subTypeName,
+                typeNode: subTypeNode,
+              })),
+              '|',
+            );
+
+            context.report({
+              node: typeNode,
+              messageId: 'objectOverridden',
+              data: {
+                subType: mergedSubTypeName,
+                superType: superTypeName,
+              },
+            });
```

Comment:
[Refactor] The two added `for (const [...` -> `context.report()` areas have very similar code around them. I was able to refactor to a shared `reportOverriddenTypes` function like:

```ts
function reportOverriddenTypes(
  overriddenObjectTypes: Map<TSESTree.TypeNode, TypeRelation[]>,
  messageId: MessageId,
  operator: '&' | '|',
) {
  for (...) {
```

WDYT?